### ISSUE TITLE
symbolic links unsupported in archives

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#83]
It looks like the engine does not support symbolic links in archives, Xonotic makes use of that, for example:

```
Warn: R_LoadDDSImage: invalid dds header "dds/textures/trak5x/floor/floor_floor2a.dds" 
Warn: R_FindImageFile could not find image 'textures/trak5x/floor/floor_floor2a.tga' in shader 'textures/trak5x/floor-floor2a' 
```

The `dds/textures/trak5x/floor/floor_floor2a.dds` is in fact a symbolic link to a file named `floor_floor2a_gloss.dds` in the same directory.

**Comment by slipher** [comment]
I would rather not support such a thing :) Is it impossible to simply fix the assets to use the real path?

**Comment by illwieckz** [comment]
These are asset paths within standard paks Xonotic ships, it means third-party community content rely on this…

**Comment by illwieckz** [comment]
Well, they can write `.shader` files to provide alternative names instead of doing symlink: every related piece of software knows how to read `.shader` files, that's not true for symlinks.

**Comment by illwieckz** [comment]
For information, there is almost 2K symlinks in xonotic paks:

```
922 xonotic-20170401-data.pk3
514 xonotic-20170401-maps.pk3
475 xonotic-20170401-nexcompat.pk3
```

**Comment by illwieckz** [comment]
There is a lot of different kind of files that are symlinked:

```
1     cfg
1349  dds
5     iqm
523   jpg
4     md3
1     mdl
28    ogg
```

**Comment by Viech** [comment]
If we restrict to relative symlinks to files within the same .dpk(dir) I don't see a (security) issue. Would that suffice for the Xonotic packs?

Allowing symlinks to files in other loaded .pdk(dir)s, if ZIP even supports that, could be doable as well but would be a little more complicated as you first need to load all packages and then evaluate the links while somehow handling the possibility that a link targets a resource that was overwritten by a package with higher priority.

**Comment by illwieckz** [comment]
See #83 for an implementation based on DarkPlaces code.

**Comment by illwieckz** [comment]
fixed by #182
